### DISCUSSION TITLE
Release Cross Fade

### DISCUSSION THREAD

**Discussion by ahall41** [GrandOrgue/grandorgue#1665] [Developers]
I'm trying to understand this ... and have a few issues which perhaps someone could explain. First of all, exactly what Pipe999ReleaseCrossfadeLength is supposed to achieve?

Having transformed quite a few HW xml ODFs for GrandOrgue, the end result can be quite unsatisfactory. Piotr Grabowski's sets typically codes 100 (all pjpes the same), which seems to work quite well. Others have a smaller value (3 or 4) which does not work well at all. I've experimented with 0 (please explain what GO does with this), 100, 200 etc, with mixed results.

What I find curious is that, if I only load a single release [I assume this is the longest?] I get a better experience (on some sets) than loading the releases and experimenting with ReleaseCrossfadeLength. 

I'm not sure if this is the same as #306 or something else. The effect is when playing staccato, the cut off seems very dead compared to a slightly less staccato note. Which I don't get if I use external reverb. I plan to do a bit more investigation, but I would appreciate some pointers as to what I should be looking for or trying out.

**Comment by midimusic** [comment]
@ahall41 

I cannot help with ODF's but can say the cross fade must be activated every time you release a key, This prevents a glitch when the release waveform begins to play. A cross fade of 100ms is normal and should work OK for all playing styles and sample sets.

csw900

**Reply by ahall41** [reply]
Thanks. I've now come to the conclusion that 0 (use GO's default - whatever that is) is better than some low number (e.g. 3 or 4) which _appears_ to be configured in the HW XML - but maybe that means something else ...

**Reply by larspalo** [reply]
The default value of the release crossfade length is not fixed. It's calculated on a case to case basis to get a good release alignment. Usually, setting the value to anything over 120 is severely affecting the release transients in a negative way. The release crossfade as such is the time to change from the current sustain section into a selected release tail.

**Comment by ahall41** [comment]
Thank you @larspalo. I think this answers my question - I will set  this to 0 in future.
